compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -325,13 +328,15 @@ impl<'interner> Monomorphizer<'interner> {
 
             HirExpression::Cast(cast) => ast::Expression::Cast(ast::Cast {
                 lhs: Box::new(self.expr(cast.lhs)),
-                r#type: Self::convert_type(&cast.r#type),
+                r#type: self.convert_type(&cast.r#type),
                 location: self.interner.expr_location(&expr),
             }),
 
             HirExpression::For(for_expr) => {
+                self.is_range_loop = true;
                 let start = self.expr(for_expr.start_range);
                 let end = self.expr(for_expr.end_range);
+                self.is_range_loop = false;
```

Comment:
I'm worried this check is a bit ad-hoc in that if the range expression contains other expressions, those would all be defaulted to `u64` as well. E.g. in `for _ in 0 .. { let x = y*10000000; 27 } { ... }`.

This is contrived though, I doubt there are any real programs that have any reason to do this.

If we ever add the `Integer` type variable kind as in https://github.com/noir-lang/noir/issues/1991 that would fix this more cleanly I think. Then any integer using comparisons or in the range of a for loop can be defaulted to u64 automatically.

Reply:
#1991 is merged as completed, should we open up another issue for this?